godal.go

```diff
@@ -4099,6 +4099,56 @@ func (ds *Dataset) SetGCPs(GCPList []GCP, opts ...SetGCPsOption) error {
 	return nil
 }
 
+// Convert list of GCPs to a GDAL GeoTransorm array
+func GCPsToGeoTransform(GCPList []GCP) (*[6]float64, error) {
+	// Convert `[]GCP` -> `C.goGCPList`
+	var gcpList C.goGCPList
+	var (
+		ids       = make([]string, len(GCPList))
+		infos     = make([]string, len(GCPList))
+		gcpPixels = make([]float64, len(GCPList))
+		gcpLines  = make([]float64, len(GCPList))
+		gcpXs     = make([]float64, len(GCPList))
+		gcpYs     = make([]float64, len(GCPList))
+		gcpZs     = make([]float64, len(GCPList))
+	)
+	for i, g := range GCPList {
+		ids[i] = g.pszId
+		infos[i] = g.pszInfo
+		gcpPixels[i] = (g.dfGCPPixel)
+		gcpLines[i] = (g.dfGCPLine)
+		gcpXs[i] = (g.dfGCPX)
+		gcpYs[i] = (g.dfGCPY)
+		gcpZs[i] = (g.dfGCPZ)
+	}
+	cIds := sliceToCStringArray(ids)
+	defer cIds.free()
+	cInfos := sliceToCStringArray(infos)
+	defer cInfos.free()
+
+	gcpList.pszIds = cIds.cPointer()
+	gcpList.pszInfos = cInfos.cPointer()
+	gcpList.dfGCPPixels = cDoubleArray(gcpPixels)
+	gcpList.dfGCPLines = cDoubleArray(gcpLines)
+	gcpList.dfGCPXs = cDoubleArray(gcpXs)
+	gcpList.dfGCPYs = cDoubleArray(gcpYs)
+	gcpList.dfGCPZs = cDoubleArray(gcpZs)
+
+	gt := make([]C.double, 6)
+	cgt := (*C.double)(unsafe.Pointer(&gt[0]))
+	if C.godalGCPListToGeoTransform(gcpList, C.int(len(GCPList)), cgt) == 0 {
```

Comment:
your error handling here is suboptimal. please use `godalWrap`/`cctx` in your c++ wrapper, and `createCgoWrapper` in your go code in order to use the standard error wrapping functionality provided by godal, as is done by all other wrapping code which is prone to returning an error

Reply:
Thanks for taking a look at this. I updated the error handling to use the `createCGOContext` to create the cctx which I then use for error handling. I also followed the convention that other functions use for creating a Opts struct that is passed in with the error handler. Could you please take a look and see if I used this paradigm correctly?

Reply:
@andyzimmerman1 lgtm thanks for the fixes. please return a [6]float64 instead of a *[6]float64 (unless there's a specific reason you chose to return a pointer, if so please explain)

Reply:
@tbonfort The only reason I was returning the pointer so that I could return `nil` for that return value in the case of an error. I though that was a standard convention, and noticed that pattern on other functions, such as `NewFeature`. I am happy to change it, though, as it is easy enough to instantiate the `[6]float64` for the error case